### PR TITLE
Typescript integration

### PR DESCRIPTION
**Pull request by Tushar-4781**
This PR corresponds to #237. There are two problems I m facing: - 

1. There is a problem with the imports. Although I did try to add tsconfigpaths in tsconfig.json still it's showing a problem. Below is the snapshot. This problem is common for all the imports
image

2. And another one is this. Like how do I define the type for this RecoilState 
image

### CODE REVIEW

**Comment by Tushar-4781** [comment]
Kindly ignore the convention problems. It's not detected by the eslint. I have to check that part. Help me with the above 2 problems

File: src/Interfaces/ILanguage.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export interface ILanguage {
+    sno: Number;
+    title: string;
+    langId: string;
+}
+
+export interface ILanguageListProps{
```
Comment (pulkitvyas08) [inline src/Interfaces/ILanguage.ts:7]: Add a space between `ILanguageListProps` and `{`

File: src/components/languagechoices/LangItem.tsx [hunk 2]
```diff
@@ -1,25 +1,25 @@
-/* eslint-disable react/prop-types */
 import React from 'react';
 import { Button } from 'react-bootstrap';
 import { useRecoilState } from 'recoil';
 import i18n from 'i18next';
 
+import { ILanguage } from 'Interfaces/ILanguage';
```
Comment (pulkitvyas08) [inline src/components/languagechoices/LangItem.tsx:6]: This should be a global import using path aliases

**Review by pulkitvyas08 — CHANGES_REQUESTED** [review]

**Comment by pulkitvyas08** [comment]
> This PR corresponds to #237. There are two problems I m facing: -
> 
> 1. There is a problem with the imports. Although I did try to add tsconfigpaths in tsconfig.json still it's showing a problem. Below is the snapshot. This problem is common for all the imports
>    image

This is not a real problem, these are just warnings, even I'm trying to get these fixed but to no avail as of now. I will create a dedicated issue for this later

> 2. And another one is this. Like how do I define the type for this RecoilState
>    image

For every recoil state, you need to define proper types. Go through the files in the recoil state and let me know if you need help with any particular one. Most of the states are pretty straightforward but in a few cases, you would have to change the structure of the states. Like in the case of `ThemeSelectionState` you would have to either go ahead and set the state to either string only or boolean only or write an interface to allow for both

**Comment by pulkitvyas08** [comment]
Also I see a lot of conflicts that cannot be resolved from github's interface, have fun 😁

File: src/Interfaces/ILanguage.ts (hunk 1 above)
Comment (Tushar-4781) [reply src/Interfaces/ILanguage.ts:7]: will do. FYI I said *Kindly ignore the convention problems. It's not detected by the eslint. I have to check that part. Help me with the above 2 problems*

File: src/components/languagechoices/LangItem.tsx (hunk 2 above)
Comment (Tushar-4781) [reply src/components/languagechoices/LangItem.tsx:6]: path aliases ain't working that's why i have not used in the interfaces as well

**Comment by Tushar-4781** [comment]
> Also I see a lot of conflicts that cannot be resolved from github's interface, have fun 😁

Umm, I can deal with the conflicts that are not the main issue. I will fix the conflicts by end of the weekend but the condition is that no one should be working on any new feature. What I mean is that once the code is 100% converted into TypeScript it can be merged then others can start working on that instead of working in JavaScript. What do you say @tijlleenders? Is it ok?

**Comment by Tushar-4781** [comment]
@pulkitvyas08, You need to convert the MyFeelings feature into TS ( including the API calls and dexi database part)

I almost resolved all path aliases problems, but image imports are still showing warnings.

**Comment by pulkitvyas08** [comment]
> @pulkitvyas08, You need to convert the MyFeelings feature into TS ( including the API calls and dexi database part)

Yes let's create a separate issue to deal with that plus TS compilation (that you haven't done, which is why the website is working without resolving all the TS related errors)

> I almost resolved all path aliases problems, but image imports are still showing warnings.

That is great to know, solves a lot of my problems

**Review by pulkitvyas08 — APPROVED** [review]

**Comment by pulkitvyas08** [comment]
> > Also I see a lot of conflicts that cannot be resolved from github's interface, have fun 😁
> 
> Umm, I can deal with the conflicts that are not the main issue. I will fix the conflicts by end of the weekend but the condition is that no one should be working on any new feature. What I mean is that once the code is 100% converted into TypeScript it can be merged then others can start working on that instead of working in JavaScript. What do you say @tijlleenders? Is it ok?

In that case please go ahead and resolve these merge conflicts to get this PR merged. I hope the process doesn't create duplicate history though

File: src/components/myfeelingspage/AddFeelings.tsx [hunk 3]
```diff
@@ -4,8 +4,13 @@ import { useTranslation } from 'react-i18next';
 import { useRecoilValue } from 'recoil';
 
 import { HeaderDashboard } from '@components/dashboard/HeaderDashboard';
+<<<<<<< HEAD:src/components/myfeelingspage/AddFeelings.tsx
```
Comment (pulkitvyas08) [inline src/components/myfeelingspage/AddFeelings.tsx]: Why are these conflict resolution marks here?

**Review by pulkitvyas08 — CHANGES_REQUESTED** [review]

File: src/components/myfeelingspage/AddFeelings.tsx (hunk 3 above)
Comment (Tushar-4781) [reply src/components/myfeelingspage/AddFeelings.tsx]: I m surprised. I resolved them at my local machine and then pushed. The working tree is clear, no conflicts are At my end

**Review by pulkitvyas08 — APPROVED** [review]

**Comment by Tushar-4781** [comment]
Resolves #237

**Pull request merged**